File: LeetCode-Java/Climbing Stairs.java
public class Solution {
    public int climbStairs(int n) {
        int[] result = new int[n + 1];
        for(int i = 0;i <= n;i ++)
        {
            if(i <= 1) result[i] = 1;
            else result[i] = result[i -1] + result[i - 2];
        }
        return result[n];
    }
}

File: Leetcode-cpp/ClimbingStairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        vector<int> record;
        record.push_back(1);
        record.push_back(2);
        for(int i = 2;i < n;i ++)
        {
            record.push_back(record[i-1] + record[i-2]);
        }
        return record[n - 1];
    }
};
